### PR TITLE
Notifications: Convert to classes

### PR DESCRIPTION
**Pull request by Maffooch**
This has been a long time coming 😄 

[sc-8894]

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

**Review by blakeaowens — APPROVED** [review]

**Review by cneill — COMMENTED** [review]
Just a couple small things

File: dojo/notifications/helper.py [hunk 1]
```diff
@@ -20,367 +22,842 @@
...
-@dojo_async_task
-@app.task
-def send_slack_notification(event, user=None, *args, **kwargs):
-    from dojo.utils import get_system_setting
+        try:
+            notification_message = render_to_string(template, kwargs)
+            logger.debug("Rendering from the template %s", template)
+        except TemplateDoesNotExist as e:
+            # In some cases, template includes another templates, if the interior one is missing, we will see it in "specifically" section
+            logger.debug(
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/notifications/helper.py:152]: I think this one should stay as debug. This log is purely for developers, and would pollute production logs with a level that high

File: dojo/notifications/helper.py [hunk 2]
```diff
@@ -20,367 +22,842 @@
...
+                notifications=self.system_notifications,
+                **kwargs,
+            )
+            # All admins will also receive system notifications, but as part of the person global notifications section below
+            # This time user is set, so will trigger email to personal email, to personal slack channel (mention), etc.
+            # only retrieve users which have at least one notification type enabled for this event type.
+            logger.debug("creating personal notifications for event: %s", event)
+            # There are notification like deleting a product type that shall not be sent to users.
+            # These notifications will have the parameter no_users=True
+            if kwargs.get("no_users") is False:
```
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/notifications/helper.py:638]: Will need to do more testing on this one. There is unit test coverage here for the way that it is

**Review by hblankenship — APPROVED** [review]

File: dojo/notifications/helper.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Maffooch) [reply dojo/notifications/helper.py]: I remembered that in most cases, this value is set to `False` in the call [here](https://github.com/Maffooch/django-DefectDojo/blob/7c71bedc62b4e548680658dda1bc7cfc95f547ac/dojo/notifications/helper.py#L54) - this where all notifications are created from within the app. The only exception is calling some of these classes directly within unit tests

To address your comment, I went with this approach: `if kwargs.get("no_users", False) is False`

**Review by cneill — APPROVED** [review]

**Pull request merged**